framework/source/class/qx/io/MDynamicScriptLoader.js

```diff
@@ -0,0 +1,218 @@
+
+   Authors:
+     * Dietrich Streifert (level420)
+
+************************************************************************ */
+
+
+/**
+ * Dynamically load non qx scripts on first instance init
+ *
+ * Usage example:
+ * 
+ * <code>
+ *  ... assets ...
+ * /**
+ *  * @asset(myapp/jquery/*)
+ *  * @asset(myapp/highcharts/*)
+ *  *
+ *  * @ignore(jQuery.*)
+ *  * @ignore(Highcharts.*)
+ *  ...
+ *
+ *  
+ *  qx.Class.define("myapp.myClass.Chart", {
+ *    extend : myapp.myClass,
+ *
+ *    include : [
+ *      qx.io.MDynamicScriptLoader
+ *    ],
+ * 
+ *    construct : function()
+ *    {
+ *      this.base(arguments);
+ * 
+ *      // other useful contructor code here
+ * 
+ *      // do something usefull when all scripts are loaded
+ *      this.addListener('scriptsReady', this.__onScriptsReady, this);
+ *   
+ *      // in debug mode load the uncompressed unobfuscated scripts
+ *      var src = '';
+ *      var min = '.min';
+ *      if (qx.core.Environment.get("qx.debug")) {
+ *        src = '.src';
+ *        min = '';
+ *      }
+ *
+ *      // initialize the script loading
+ *      this._loadScriptsDynamic([
+ *        "myapp/jquery/jquery"+min+".js",
+ *        "myapp/highcharts/highcharts"+src+".js",
+ *        "myapp/highcharts/highcharts-more"+src+".js",
+ *        "myapp/highcharts/highcharts-modifications.js"
+ *      ]);
+ *    }
+ *  });
+ *      
+ * </code>
+ */
+qx.Mixin.define("qx.io.MDynamicScriptLoader",
+{
+  /*
+  *****************************************************************************
+     EVENTS
+  *****************************************************************************
+  */
+
+  events : {
+    /**
+     * fired when a specific script is loaded successfully
+     */
+    scriptLoadingSuccess: 'qx.event.type.Data',
+
+    /**
+     * fired when a specific script fails loading
+     */
+    scriptLoadingFailed: 'qx.event.type.Data',
+
+    /**
+     * fired when all given scripts are loaded, the first time all scripts reported
+     * load success
+     */
+    scriptsLoadingComplete: 'qx.event.type.Event',
+    
+    /**
+     * fired when all given scripts are loaded, each time loadScriptsDynamic is called
+     */
+    scriptsReady: 'qx.event.type.Event'
+  },
+
+
+  /*
+  *****************************************************************************
+     MEMBERS
+  *****************************************************************************
+  */
+
+
+  members : {
+  
+    /**
+     * Load scripts dynamically, typically used in the instance constructor
+     * 
+     * @param codeArr {Array} an array with the uri names of the scripts
+     * 
+     */
+    _loadScriptsDynamic: function(codeArr) {
+      this.__initStateHashes();      
+      this.__loadScriptArr(codeArr);
+    },
+
+    
+    /**
+     * Initialize the state hashes if not yet done
+     */
+    __initStateHashes : function() {
+      if(!this.constructor.__LOADING) {

```

Comment:
@derrell: this does not work in mixins. If i declare a statics section the mixin errors.